src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -400,88 +499,166 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
     }
 
     protected emitClassMembers(c: ClassType): void {
-        this.forEachClassProperty(c, "none", (name, jsonName, property) => {
+        if (this._options.withGetterSetter) {
+            this.emitLine("private:");
+            this.ensureBlankLine();
+
+            this.forEachClassProperty(c, "none", (name, _jsonName, property) => {
+                this.emitLine(this.cppType(property.type, { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false }, true), " ", name, ";");
+            });
+
+            this.emitLine("public:");
+            this.ensureBlankLine();
+        }
+
+        this.forEachClassProperty(c, "leading-and-interposing", (name, jsonName, property) => {
             this.emitDescription(this.descriptionForClassProperty(c, jsonName));
-            this.emitLine(
-                this.cppType(
-                    property.type,
-                    { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false },
-                    true
-                ),
-                " ",
-                name,
-                ";"
-            );
+            if (!this._options.withGetterSetter) {
+                this.emitLine(this.cppType(property.type, { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false }, true), " ", name, ";");
+            } else {
+                const [getterName, mutableGetterName, setterName] = defined(this._gettersAndSettersForPropertyName.get(name));
+                const rendered = this.cppType(property.type, { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false }, true);
+                this.emitLine("const ", rendered, " & ", getterName, "() const { return ", name, "; }");
+                this.emitLine(rendered, " & ", mutableGetterName, "() { return ", name, "; }");
+                this.emitLine("void ", setterName, "( const ", rendered, "& value) { ", name, " = value; }");   
+            }
         });
     }
 
     protected emitClass(c: ClassType, className: Name): void {
         this.emitDescription(this.descriptionForType(c));
-        this.emitBlock(["struct ", className], true, () => {
+        this.emitBlock([ this._options.withGetterSetter ? "class " : "struct ", className], true, () => {
+            if (this._options.withGetterSetter) {
+                this.emitLine("public:");
+                this.ensureBlankLine();
+                this.emitLine(className, "() = default;");
+                this.emitLine("virtual ~", className, "() = default;");
+                this.ensureBlankLine();
+
+                this.emitLine("friend std::ostream& operator<<(std::ostream& os, ", className, " const& ms){");
```

Comment:
Will this work for all property types, including optionals, dictionaries, enums, etc...?

Reply:
Also, use `emitBlock` here.  That

1. uses consistent bracing style
2. does indentation automatically.  Indentation is not fixed in quicktype, actually - a target language can override it, so if you hard-code it, we can't do that anymore.

Reply:
Hah, great, I was pretty ashamed of the manual tabbing myself :(